lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -1830,24 +1841,91 @@ namespace Js
 
             Var* moduleMemoryPtr = RecyclerNewArrayZ(GetRecycler(), Var, wasmModule->memSize);
 
             Var* heap = moduleMemoryPtr + wasmModule->heapOffset;
 
             exportObj = JavascriptOperators::NewJavascriptObjectNoArg(this);
-            Js::Var exportsNamespace = JavascriptOperators::NewJavascriptObjectNoArg(this);
 
-            PropertyRecord const * exportsPropertyRecord = nullptr;
+            Var* localModuleFunctions = moduleMemoryPtr + wasmModule->funcOffset;
 
+            FrameDisplay * frameDisplay = RecyclerNewPlus(GetRecycler(), sizeof(void*), FrameDisplay, 1);
+            frameDisplay->SetItem(0, moduleMemoryPtr);
+            bool hasAnyLazyTraps = false;
+            const auto createLazyTrap = [this, &exportObj]() {
+                JavascriptLibrary *library = this->GetLibrary();
+                JavascriptError *pError = library->CreateError();
+                JavascriptExceptionObject * exceptionObject =
+                    RecyclerNew(this->GetRecycler(), JavascriptExceptionObject, exportObj, this, NULL);
+                pError->SetJavascriptExceptionObject(exceptionObject);
+                return library->CreateStdCallExternalFunction((Js::StdCallJavascriptMethod)WasmLazyTrapCallback, 0, pError);
+            };
+            for (uint i = 0; i < wasmModule->funcCount; ++i)
+            {
+                if (functionArray[i] == nullptr) 
+                {
+                    Assert(PHASE_ON1(WasmLazyTrapPhase));
+                    hasAnyLazyTraps = true;
+                    localModuleFunctions[i] = createLazyTrap();
+                    continue;
+                }
+                AsmJsScriptFunction * funcObj = javascriptLibrary->CreateAsmJsScriptFunction(functionArray[i]->body);
+                funcObj->GetDynamicType()->SetEntryPoint(AsmJsExternalEntryPoint);
+                funcObj->SetModuleMemory(moduleMemoryPtr);
+                FunctionEntryPointInfo * entypointInfo = (FunctionEntryPointInfo*)funcObj->GetEntryPointInfo();
+                entypointInfo->SetIsAsmJSFunction(true);
+                entypointInfo->address = AsmJsDefaultEntryThunk;
+                entypointInfo->SetModuleAddress((uintptr_t)moduleMemoryPtr);
+                funcObj->SetEnvironment(frameDisplay);
+                localModuleFunctions[i] = funcObj;
+                // Do MTJRC/MAIC:0 check
+#if ENABLE_DEBUG_CONFIG_OPTIONS
+                if (CONFIG_FLAG(MaxAsmJsInterpreterRunCount) == 0)
+                {
+                    GenerateFunction(GetNativeCodeGenerator(), funcObj->GetFunctionBody(), funcObj);
+                }
+#endif
+            }
+
+            Js::Var exportsNamespace = nullptr;
+
+            // Check for Default export
+            for (uint32 iExport = 0; iExport < wasmModule->info->GetExportCount(); ++iExport)
+            {
+                Wasm::WasmExport* funcExport = wasmModule->info->GetFunctionExport(iExport);
+                if (funcExport && funcExport->nameLength == 0)
```

Comment:
I don't think this is specified in the spec. At this time this is a convenience for testing.
If you have a default export and normal exports, we will attach on the default function.
```js
const m = Wasm.instantiateModule(convertToBinary(`(module (func (result i32) (i32.const 1)) (func (result i32) (i32.const 2)) (export "" 0) (export "t" 1))`), {});
print(m.exports());
print(m.exports.t());
```
will print
```
1
2
```